sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectView.groovy

```diff
@@ -354,4 +309,69 @@ class ProjectView extends ProjectDesign {
         DataProvider<ProjectSummary, ?> dataProvider = this.projectGrid.getDataProvider()
         return dataProvider
     }
+
+    /**
+     * Adds responsiveness to an abstractComponent
+     *
+     * <p>This applies the css class style .responsive-grid-layout to the provided abstractComponent allowing it to display it's content in a responsive manner</p>
+     *
+     * @param AbstractComponent the {@link com.vaadin.ui.AbstractComponent}, where the css style and responsiveness should be added
+     * @since 1.0.2
+     */
+    static void setupLayoutResponsiveness(AbstractComponent abstractComponent) {
+        abstractComponent.addStyleName("responsive-grid-layout")
+        abstractComponent.setWidthFull()
+    }
```

Comment:
no static please, this needs to be done with a default setting upon object construction. The interface methods then allow for a change in the behaviour. For example, the default could be non-responsive